Modules/_tkinter.c

```diff
@@ -261,6 +261,10 @@ static PyThreadState *tcl_tstate = NULL;
 
 #define ENTER_PYTHON \
     { PyThreadState *tstate = tcl_tstate; tcl_tstate = NULL; \
+        if(tstate == NULL) { \
+          PyGILState_STATE gstate = PyGILState_Ensure(); \
+          tstate = PyThreadState_Get(); \
+          PyGILState_Release(gstate); } \
```

Comment:
Won't this release the GIL even if it was previously held?

Reply:
The problem arises because the GIL has never seen the thread before (so the thread state is NULL) - if the GIL was already acquired, the thread state would not be NULL so we wouldn't be calling this (I believe, though I assume this is not the case given that you are asking).

Reply:
> Won't this release the GIL even if it was previously held?

No, it will not release the GIL if it was already held before calling [PyGILState_Ensure](https://docs.python.org/3/c-api/init.html#c.PyGILState_Ensure).